File: 백준/1000/1416/1416.js
const fs = require("fs");
const stdin = (
  process.platform === "linux"
    ? fs.readFileSync("/dev/stdin").toString()
    : `6 2
    3 4 5 6 11 -1`
).split("\n");

const input = (() => {
  let line = 0;
  return () => stdin[line++];
})();

const [N, M] = input()
  .trim()
  .split(" ")
  .map((v) => +v);

const arr = input()
  .trim()
  .split(" ")
  .map((v) => +v);
console.log(arr);

File: 백준/1000/1416/1416.py


from bisect import bisect_left
import sys
input = sys.stdin.readline

N, M = map(int, input().strip().split(' '))
arr = list(map(lambda x: x*2, map(int, input().strip().split(' '))))
arr.sort()
idx = bisect_left(arr, 0)
arrMin = arr[:idx]
arrMax = arr[idx:][::-1]
result = 0
# print(N, M, arrMin, arrMax)
for i in range(0, len(arrMin), M):
    result += abs(arrMin[i])
for i in range(0, len(arrMax), M):
    result += abs(arrMax[i])

result -= (max(abs(arrMin[0] if arrMin else 0),
           abs(arrMax[0] if arrMax else 0))//2)
print(result)
